staging/src/k8s.io/client-go/tools/clientcmd/validation.go

```diff
@@ -179,10 +179,6 @@ func validateClusterInfo(clusterName string, clusterInfo clientcmdapi.Cluster) [
 			validationErrors = append(validationErrors, fmt.Errorf("no server found for cluster %q", clusterName))
 		}
 	}
-	// Make sure CA data and CA file aren't both specified
-	if len(clusterInfo.CertificateAuthority) != 0 && len(clusterInfo.CertificateAuthorityData) != 0 {
```

Comment:
if there's a documented precedence between these two fields, this could be worth a warning but probably should not be an error, and the data field should take precedence as documented (and not cause an open/read of the file). same comment applies to the other file/data field pairs.

Reply:
alternately, we could look into updating the way the [string flags bind](https://github.com/kubernetes/kubernetes/blob/190f6d870ea396f27145f07fc00aa784b65358ef/staging/src/k8s.io/client-go/tools/clientcmd/overrides.go#L227) so that setting them sets the cert file field *and* clears the data field. that would involved a new type of flag whose Set implementation sets one field and clears another